src/services/swap/swap.service.ts

```diff
@@ -0,0 +1,244 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { exchangeProxyService } from '../contracts/exchange-proxy.service';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { Swap } from '@balancer-labs/sor/dist/types';
+import { AddressZero } from '@ethersproject/constants';
+import { NATIVE_ASSET_ADDRESS } from '@/constants/tokens';
+import { getWstETHByStETH, isStETH } from '@/lib/utils/balancer/lido';
+import { SwapV2 } from '@balancer-labs/sor2';
+import ConfigService, { configService } from '@/services/config/config.service';
+import { vaultService } from '@/services/contracts/vault.service';
+import { lidoRelayerService } from '@/services/contracts/lido-relayer.service';
+import {
+  FundManagement,
+  SingleSwap,
+  SwapKind
+} from '@balancer-labs/balancer-js';
+import Web3Service, { web3Service } from '../web3/web3.service';
+
+export type Address = string;
+
+export enum SwapTokenType {
+  fixed,
+  min,
+  max
+}
+
+export interface SwapToken {
+  address: Address;
+  amount: BigNumber;
+  type: SwapTokenType;
+}
+
+export default class SwapService {
+  constructor(
+    private readonly config: ConfigService = configService,
+    private readonly web3: Web3Service = web3Service
+  ) {}
+
+  public async batchSwapV1(
+    tokenIn: SwapToken,
+    tokenOut: SwapToken,
+    swaps: Swap[][]
+  ): Promise<TransactionResponse> {
+    console.log('[Swap Service] batchSwapV1');
+
+    const options: any = {};
+
+    /* I have no idea what this is doing still... */
+    if (tokenIn.address === NATIVE_ASSET_ADDRESS) {
+      options.value = `0x${tokenIn.amount.toHexString()}`;
+    }
+
+    try {
+      return exchangeProxyService.multihopBatchSwap(
+        swaps,
+        tokenIn,
+        tokenOut,
+        options
+      );
+    } catch (e) {
+      console.log('[Swapper] batchSwapV1 Error:', e);
+      return Promise.reject(e);
+    }
+  }
+
+  public async batchSwapV2(
+    tokenIn: SwapToken,
+    tokenOut: SwapToken,
+    swaps: SwapV2[],
+    tokenAddresses: string[]
+  ): Promise<TransactionResponse> {
+    if (isStETH(tokenIn.address, tokenOut.address)) {
+      return this.lidoBatchSwap(tokenIn, tokenOut, swaps, tokenAddresses);
+    }
+    console.log('[Swap Service] batchSwapV2');
+    const overrides: any = {};
+
+    if (tokenIn.address === AddressZero) {
```

Comment:
I'm interested why this is AddressZero (0x000...) and for the v1 batch swap we're using the native asset address which is (0xEeee...). Please test to see if tokenIn.address for a v2 batch swap will actually be 0x00.. when ETH is the token in and not 0xEeee...

Reply:
Yea I'm not sure why it's like this, it was like this in Swapper so left it as is.